### PR TITLE
Upgrade webhook protocol

### PR DESCRIPTION
**Pull request by fabxc**
We basically want the same info for webhook integrations as for templating.
This will also make maintenance easier since we don't have two separate formats we have to version.

@brian-brazil

### CODE REVIEW

**Comment by beorn7** [comment]
Note: tests are failing.

File: template/template.go [hunk 1]
```diff
@@ -205,24 +205,25 @@ func (kv KV) Values() []string {
...
 type Alert struct {
-	Status       string
-	Labels       KV
-	Annotations  KV
-	WasSilenced  bool
-	WasInhibited bool
+	Status       string `json:"status"`
+	Labels       KV     `json:"labels"`
+	Annotations  KV     `json:"annotations"`
+	WasSilenced  bool   `json:"-`
```
Comment (brian-brazil) [inline template/template.go]: Missing a " ?

File: template/template.go [hunk 2]
```diff
@@ -205,24 +205,25 @@ func (kv KV) Values() []string {
 // as this will confuse them and prevent simple things like
 // simple equality checks to fail. Map everything to float64/string.
 type Data struct {
```
Comment (brian-brazil) [inline template/template.go]: We should add a note here that this is also going to the webhook. Sending things over the network is more expensive than rendering them in memory.

Comment (fabxc) [inline template/template.go]: The whole point of AM is that there are only very few notifications. So I'm not worried about any network load.
Adding comment anyway.

**Comment by fabxc** [comment]
Updated

**Comment by brian-brazil** [comment]
:+1:

File: template/template.go (hunk 2 above)
Comment (brian-brazil) [inline template/template.go]: "Few" is a relative term. It's not hard to imagine a situation where you'd have thousands of alerts per second worth of webhook firing.

Comment (fabxc) [inline template/template.go]: Honestly, we are worrying about 1000x scale way to often for Alertmanager at this point.
And if we reach that point, 200 more bytes of JSON will certainly not be the bottleneck.

**Pull request merged**